### PR TITLE
Change `logger` to study's module variable

### PR DESCRIPTION
**Pull request by crcrpar**
Personally, there is no specific reason for `Study` to have `logger` as its attribute and the below code block does not look solid.

https://github.com/optuna/optuna/blob/dab63878bf348b16713df23691c2fe9bb467dcdf/optuna/study.py#L172-L185

### CODE REVIEW

File: optuna/study.py [hunk 1]
```diff
@@ -44,6 +44,9 @@
     ObjectiveFuncType = Callable[[trial_module.Trial], float]
 
 
+logger = logging.get_logger(__name__)
```
Comment (hvy) [inline optuna/study.py]: How about marking the object private by prefixing it with an underscore, as it's not very likely to be used from outside this submodule?

Comment (crcrpar) [reply optuna/study.py]: makes sense.

I didn't know that optuna/integration/sklearn.py does that.

Comment (hvy) [reply optuna/study.py]: Me neither, until you mentioned. 😅

**Comment by hvy** [comment]
@sile, sorry for interrupting, could you go ahead and merge this PR if it still looks okay?

**Comment by sile** [comment]
@hvy, @crcrpar Thanks! I also think that making the logger private is a good change. I'll merge this PR.

**Pull request merged**